common/app/routes/Challenges/Child-Container.jsx

```diff
@@ -1,20 +1,56 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { createSelector } from 'reselect';
+
+import { challengeSelector } from '../../redux';
+import { challengeUpdated } from './redux';
 
 import CompletionModal from './Completion-Modal.jsx';
 import AppChildContainer from '../../Child-Container.jsx';
+import { OverlayLoader } from '../../helperComponents';
+
+const mapStateToProps = createSelector(
+  challengeSelector,
+  challenge => {
+    const { description } = challenge;
+    return {
+      challenge,
+      showLoading: !description || description.length === 0
```

Comment:
This make the loading script brittle, as we cannot guarantee every challenge in the future will have a description. Why not use _.empty?

Reply:
I need to use some property of a challenge that will always be there for a rendered challenge, but will not be available if only the mapUi data is available. `description` seemed like the most sensible option at the time. 

Do you have another suggestion?